oval_graph/client.py

```diff
@@ -105,29 +109,54 @@ def create_dict_of_rule(self, rule_id):
             return converter.to_JsTree_dict()
         return converter.to_sigma_dict(0, 0)
 
-    def save_dict(self, dict):
-        with open(self.xml_parser.get_src(self.html_interpreter + '/data.js'), "w+") as data_file:
+    def save_dict(self, dict_, rule, date):
+        with open(os.path.join(tempfile.gettempdir(),
+                               'graph-of-' + rule +
```

Comment:
This dirname construction repeats in several places. Why not to have a function that will create the `empfile.gettempdir() / 'graph-of-' + rule + date` directory when you supply `rule` and `date`? Also check the `pathlib` module, it enables you to construct paths using the / operator without having to call `os.path.join()` function.